src/data-channel-wrapper.cpp

```diff
@@ -48,21 +48,25 @@ DataChannelWrapper::DataChannelWrapper(const Napi::CallbackInfo &info) : Napi::O
 
     mDataChannelPtr->onClosed([&]()
                               {
-        if (mOnClosedCallback)
+        if (mOnClosedCallback){
             mOnClosedCallback->call([this](Napi::Env env, std::vector<napi_value> &args) {
                 // Check the peer connection is not closed
                 if(instances.find(this) == instances.end())
                     throw ThreadSafeCallback::CancelException();
 
-                cleanCbsAndEraseInstance();
+                instances.erase(this);
```

Comment:
The problem is that erasing the instance is actually what prevents already-scheduled callbacks to run. Keeping it here means scheduled callbacks may still be run between `close()` and `onClose`.